### PR TITLE
bpo-42756: Configure LMTP Unix-domain socket to use global default timeout when timeout not provided

### PR DESCRIPTION
**Pull request by rrhodes**
What?
=====
1. Configure LMTP Unix-domain socket to use socket global default timeout when a timeout is not explicitly provided.
2. Removed unused "int_port" from the mock socket `create_connection` method.

Why?
====
To allow users to apply the default timeout without explicitly needing to pass in `None`.

https://bugs.python.org/issue42756

### CODE REVIEW

File: Lib/test/test_smtplib.py [hunk 1]
```diff
@@ -165,12 +165,28 @@ class LMTPGeneralTests(GeneralTests, unittest.TestCase):
...
+        self.assertIsNone(client.sock.gettimeout())
+        client.close()
+
     def testTimeoutZero(self):
         super().testTimeoutZero()
         local_host = '/some/local/lmtp/delivery/program'
         with self.assertRaises(ValueError):
             self.client(local_host, timeout=0)
 
+
```
Comment (corona10) [inline Lib/test/test_smtplib.py]:
```suggestion
```

File: Lib/test/mock_socket.py [hunk 2]
```diff
@@ -107,16 +107,16 @@ def getpeername(self):
...
         pass
 
+    def connect(self, host):
+        pass
+
 
 def socket(family=None, type=None, proto=None):
     return MockSocket(family)
 
+
```
Comment (corona10) [inline Lib/test/mock_socket.py]:
```suggestion
```

File: Lib/test/mock_socket.py [hunk 3]
```diff
@@ -107,16 +107,16 @@ def getpeername(self):
...
 def create_connection(address, timeout=socket_module._GLOBAL_DEFAULT_TIMEOUT,
                       source_address=None):
-    try:
-        int_port = int(address[1])
```
Comment (corona10) [inline Lib/test/mock_socket.py]: Please revert this change.
IMHO, this code was intended for checking port value validation.

**Review by corona10 — CHANGES_REQUESTED** [review]

File: Lib/smtplib.py [hunk 4]
```diff
@@ -1082,7 +1082,9 @@ def connect(self, host='localhost', port=0, source_address=None):
         # Handle Unix-domain sockets.
         try:
             self.sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
-            self.sock.settimeout(self.timeout)
+
```
Comment (corona10) [inline Lib/smtplib.py]:
```suggestion
```

**Comment by corona10** [comment]
@vstinner Can you please take a look.
This PR is following what socket.create_connection doing.
https://github.com/python/cpython/blob/c8a7b8fa1b5f9a6d3052db792018dc27c5a3a794/Lib/socket.py#L827

File: Lib/test/test_smtplib.py [hunk 5]
```diff
@@ -165,12 +165,28 @@ class LMTPGeneralTests(GeneralTests, unittest.TestCase):
 
     client = smtplib.LMTP
 
+    def testUnixDomainSocketTimeoutDefault(self):
```
Comment (corona10) [inline Lib/test/test_smtplib.py:169]: We can use @unittest.skipUnless ;)

**Review by corona10 — APPROVED** [review]

**Comment by corona10** [comment]
@rrhodes Thank you for working on this!

**Comment by corona10** [comment]
@rrhodes Would you like to submit the backport patch for this?

**Comment by rrhodes** [comment]
> @rrhodes Would you like to submit the backport patch for this?

Sure, first time backporting so bear with me please.

**Comment by corona10** [comment]
@rrhodes 
Using the cherry-picker is the easy way

https://pypi.org/project/cherry-picker/#about
